sopel/irc/__init__.py

```diff
@@ -840,18 +845,22 @@ def say(
                     time.sleep(sleep_time)
 
             # Loop detection
-            messages = [m[1] for m in recipient_stack['messages'][-8:]]
+            if antiloop_threshold > 0 and elapsed < antiloop_window:
+                messages = [m[1] for m in recipient_stack['messages'][-10:]]
```

Comment:
I realize it would obsolete #2327 before it even makes it out of draft status (and make more work for myself in the documentation that I just reworded above), but what if Sopel maintained a time-based rather than count-based history of messages it has sent?

As I said in one of those rephrasing line notes, the `antiloop_window` becomes useless if the bot has sent too many messages to a given channel, because the 10-slot history may be far too small for the volume of output being generated.

Reply:
This will be for another time. I think the list should contain up to N messages, N being configurable, as long as it is up to the last T seconds, for a maximum size of B bytes.

This is, I think, for another good PR or two.

Reply:
Let me add a note to that other PR, then, and resolve this thread.